source/fe/fe_q_base.cc

```diff
@@ -19,8 +19,10 @@
 #include <deal.II/base/template_constraints.h>
 #include <deal.II/base/tensor_product_polynomials.h>
 #include <deal.II/base/tensor_product_polynomials_const.h>
+#include <deal.II/base/tensor_product_polynomials_bubbles.h>
 #include <deal.II/base/polynomials_piecewise.h>
 #include <deal.II/fe/fe_q_base.h>
+#include <deal.II/fe/fe_q_bubbles.h>
```

Comment:
I don't like to include a derived class at this point (you need it for dynamic_cast in the code below). Since all appearances go over the polynomial, I'd rather check `types_are_equal<POLY,TensorProductPolynomialsBubble<dim>>::value` (from base/template_constraints.h).